### PR TITLE
make llm and embeddings more optional and robust

### PR DESCRIPTION
**Pull request by logan-markewich**
# Description

If a user REALLY doesn't want to use embeddings or an LLM, you can now set them to `None` explicitly. 

Furthermore, resolving the `embed_model` has been improved to use appropriate classes for most huggingface models.

## Type of Change

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?

- [x] I stared at the code and made sure it makes sense

### CODE REVIEW

File: llama_index/llms/utils.py [hunk 1]
```diff
@@ -6,13 +6,27 @@
...
+    if llm == "default":
+        # return default OpenAI model. If it fails, return LlamaCPP
+        try:
+            llm = OpenAI()
+        except ValueError:
+            llm = "local"
+            print(
+                "******\n"
+                "Could not load OpenAI model. Using default LlamaCPP=llama2-13b-chat. "
+                "If you intended to use OpenAI, please check your API key."
```
Comment (yisding) [inline llama_index/llms/utils.py:26]: Does this mean that if OPENAI_API_KEY isn't set, then we default to LlamaCPP?

Comment (logan-markewich) [reply llama_index/llms/utils.py:26]: Exactly. Just a quick way to unblock people who don't want to set their openai key. This was actually introduced in 0.8.0

Pretty common issue on the discord/github "how do I get rid of OpenAI" -- now you can :)

Comment (yisding) [reply llama_index/llms/utils.py:26]: OK, maybe this message should be stronger / a more explicit opt in along with an explicit reference to OPENAI_API_KEY (is there any equivalent to a press enter to continue in notebooks?) I'm not sure if your average person will have the compute/free memory to run Llama2-13B (requires 7GB of free memory) when they are starting out.

Comment (logan-markewich) [reply llama_index/llms/utils.py:26]: It runs on CPU, unless you specifically installed llama-cpp compiled for your GPU

I can add a more explicit log though

**Pull request merged**